test/onnx/verify_onnx.cpp

```diff
@@ -1014,6 +1014,103 @@ TEST_CASE(instance_norm_3d_test)
     EXPECT(migraphx::verify::verify_rms_range(result_vector, gold));
 }
 
+TEST_CASE(isinf_half_test)
+{
+    migraphx::program p = migraphx::parse_onnx("isinf_half_test.onnx");
+    p.compile(migraphx::make_target("ref"));
+
+    migraphx::shape s{migraphx::shape::half_type, {2, 3}};
+    migraphx::parameter_map pp;
+    float nan      = std::numeric_limits<float>::quiet_NaN();
+    float infinity = std::numeric_limits<float>::infinity();
+    float x        = std::numeric_limits<float>::max();
+    // Use a synthesized value to catch any run time FP results
+    // that would bypass FP-units if compiler option like --fast-math is turned on
+    float synth_infinity   = x * x;
```

Comment:
Dont use a synthesized value here since according to @lakhinderwalia this may not produce an infinity value.

Reply:
@pfultz2: that is the intent. So the unit test should fail. 

This is exactly how those values that are produced in a graph during computation may actually not be INFINITY, and lsinf() would give false results.

Reply:
The unit test should only fail if we are not calling isinf correctly.

Checking if a synthesized value is infinity, should be done as a separate unit test although I don't see how it would be useful. We don't run these tests when running the model. And if it produced a different value for infinity it may not even effect the model accuracy either.

Reply:
> The unit test should only fail if we are not calling isinf correctly.

We are in agreement. Hence my suggestions. Thanks.

Reply:
But if you use a synthesized value then the test can fail because the synthesized value is not infinity,  not because we are calling isinf incorrectly, which is why we shouldn't use a synthesized value for this test.

Reply:
For the record, those synthesized values are all passing as infinite values in each tests on the CI.